src/templates/header.js

```diff
@@ -11,20 +11,91 @@ const links = [
   {title: 'Contact', link: '/contact'},
 ]
 
-const Header = ({ siteTitle }) => {
-  return (
-  <header>
-    <div>
-      <Nav 
-        logo={Logo({title: siteTitle})}
-        links={links}
-        GatsbyLink={Link}
-      />
-    </div>
-  </header>
-)
+class Header extends Component {
+
+  componentDidMount() {
+    if(typeof window !== 'undefined') {
+      let userDetails = {}
+      let SITE_NAME = window.location.origin;
+      document.getElementById('contact-form') &&
+      document
+        .getElementById('contact-form')
+        .addEventListener('submit', (e => {
+          e.preventDefault()
+          const idea = document.getElementById('idea').value
+          const email = document.getElementById('email').value
+          document.getElementById('idea').value = ""
+          document.getElementById('email').value = ""
+  
+          const formSubmitData = {
+            email,
+            idea,
+            userDetails,
+          }
+  
+          const url = 'https://hooks.zapier.com/hooks/catch/1041785/21ewe5/'
+
+          const http = new XMLHttpRequest();
+          http.open('POST', url)
+          http.onreadystatechange = function() {
+            if(http.readyState === 4 && http.status === 200) {
```

Comment:
It is better to add a comment on what `readyState` with value 4 means.
Or 
You may use the constants in XMLHttpRequest to make it clear. 
For example, instead of writing `http.readyState === 4` you can write like `http.readyState === XMLHttpRequest.DONE`